File: programmers/DFS_BFS/네트워크.js
function solution(n, computers) {
  function bfs(visit, i) {
    const q = [i];
    while (q.length) {
      let node = q.pop();
      visit[node] = true;
      for (let v = 0; v < n; v++) {
        if (v === node) continue;
        if (!visit[v] && computers[node][v]) {
          q.push(v);
        }
      }
    }
  }

  let ans = 0;
  const visit = Array(n).fill(false);
  for (let i = 0; i < n; i++) {
    if (visit[i] === false) {
      bfs(visit, i);
      ans += 1;
    }
  }
  return ans;
}

File: programmers/DFS_BFS/네트워크.py
def solution(n, computers):
    def bfs(visit, i):
        q = [i]
        while q:
            node = q.pop()
            visit[node] = 1
            for v in range(n):
                if v == node:
                    continue
                if not visit[v] and computers[node][v]:
                    q.append(v)
    answer = 0
    visit = [0] * n
    for i in range(n):
        if not visit[i]:
            bfs(visit, i)
            answer += 1
    return answer
